vasern/src/core/vasern/Document.d.ts

```diff
@@ -0,0 +1,124 @@
+import EventSubscriber, { Event } from "../vasern-subscriber/EventSubscriber.d";
+import RawObject from "../vasern-parser/Parser";
+import ObjectID, { NewObject } from "../../plugins/vasern-objectid";
+import ConfigProps from "../../config";
+import Queryable from "vasern/vasern/src/core/vasern-queryable";
+
+export type Args = {
+    props: any;
+    name: string;
+    version?: number;
+    assignTo?: string;
+    callbackFunction?: () => any;
+}
+
+export type NewValue = {
+    id: string;
+    rest: Object;
+}
+
+export type UpdateFunc = {
+    remove: (query: string | Object) => boolean;
+    insert: (input: Object | Object[]) => false | NewObject;
+    update: (query: string | Object, newValues: NewValue) => false | NewObject;
+    get: (query: string | Object) => NewObject;
+}
+
+export type CommitedItems = {
+    insert: NewObject[];
+    update: NewObject[];
+    remove: NewObject[];
+}
+
+export type Plugin = {
+    methods: Function[];
+    name: string;
+}
+
+// TODO: I'm not sure about all the query types as get isn't implemented
+// TODO: Also in the example data() gets called but there's no such function here?
+export default class Document extends Queryable {
```

Comment:
I have added `Queryable` as the extension. It helps typescript regconizes methods from Queryable.
Though it looks different from the .js implementation (which `Queryable` was imported - [Document > line 571](https://github.com/curtisy1/vasern/blob/e86781931852e45c1d12e5c15bea9f627b9f000f/vasern/src/core/vasern/Document.js#L571))

Reply:
Yeah makes sense now that I think about it. Also makes the TODO's obsolete since types are now inherited from Queryable directly.